### ISSUE TITLE
uni extend: seems to infer wrong temp width

### ISSUE THREAD

**Issue by matsc-at-sics-se** [matsc-at-sics-se/unison#45]
[fu2_atan2.mir.zip](https://github.com/matsc-at-sics-se/unison/files/2094773/fu2_atan2.mir.zip)
```
uni import -f fu2_atan2 fu2_atan2.mir -o fu2_atan2.uni --target=X86 --explicitcallreg --maxblocksize=25 --goal=speed --copyremat --lint
uni linearize fu2_atan2.uni -o fu2_atan2.lssa.uni --target=X86 --lint
uni extend fu2_atan2.lssa.uni -o fu2_atan2.ext.uni --target=X86 --lint
consistentOperands: in operation o123, the inferred width of temporary t193 (16) is inconsistent with corresponding register class (InfiniteRegisterClass RM64) given in the target description
consistentOperands: in operation o124, the inferred width of temporary t194 (16) is inconsistent with corresponding register class (InfiniteRegisterClass RM64) given in the target description
consistentOperands: in operation o126, the inferred width of temporary t194 (16) is inconsistent with corresponding register class (InfiniteRegisterClass RM64) given in the target description
consistentOperands: in operation o133, the inferred width of temporary t194 (16) is inconsistent with corresponding register class (InfiniteRegisterClass RM64) given in the target description
consistentOperands: in operation o168, the inferred width of temporary t269 (16) is inconsistent with corresponding register class (InfiniteRegisterClass RM64) given in the target description
consistentOperands: in operation o170, the inferred width of temporary t269 (16) is inconsistent with corresponding register class (InfiniteRegisterClass RM64) given in the target description
```
```
    o123: [t193] <- {FsFLD0SD, FsFLD0SD_source} []
    o124: [t194] <- {-, MOVE128, STORE128, FsFLD0SD_demat} [t193] (remat-origin: o123)
    o126: [t196] <- {-, MOVE128, LOAD128, FsFLD0SD_remat} [t194] (remat-origin: o123)
```

**Comment by matsc-at-sics-se** [comment]
Apparently `widthOfTemp ra cg f t (d:us)` returns 16 instead of 8, where `t = t112` before renaming temps.

**Comment by matsc-at-sics-se** [comment]
[spi_get_float.mir.zip](https://github.com/matsc-at-sics-se/unison/files/2127385/spi_get_float.mir.zip)

Likely a smaller case:
```
uni import -f spi_get_float spi_get_float.mir -o spi_get_float.uni --target=X86 --explicitcallreg --maxblocksize=25 --goal=speed --copyremat --lint
consistentOperands: in operation o17, the inferred width of temporary t14 (8) is inconsistent with corresponding register class (RegisterClass FR64) given in the target description
Makefile:7165: recipe for target 'spi_get_float.uni' failed
make: [spi_get_float.uni] Error 1 (ignored)
uni linearize spi_get_float.uni -o spi_get_float.lssa.uni --target=X86 --lint
consistentOperands: in operation o17, the inferred width of temporary t25 (8) is inconsistent with corresponding register class (RegisterClass FR64) given in the target description
Makefile:7168: recipe for target 'spi_get_float.lssa.uni' failed
make: [spi_get_float.lssa.uni] Error 1 (ignored)
uni extend spi_get_float.lssa.uni -o spi_get_float.ext.uni --target=X86 --lint
consistentOperands: in operation o53, the inferred width of temporary t66 (8) is inconsistent with corresponding register class (RegisterClass FR64) given in the target description
Makefile:7171: recipe for target 'spi_get_float.ext.uni' failed
make: [spi_get_float.ext.uni] Error 1 (ignored)
uni augment spi_get_float.ext.uni -o spi_get_float.alt.uni --target=X86 --copyremat --lint
consistentOperands: in operation o54, the inferred width of temporary t48 (8) is inconsistent with corresponding register class (RegisterClass FR64) given in the target description
```
I suspect that the trouble stems from this in the *.mir:
```
registers:       
  - { id: 15, class: fr64 }
[...]
    %xmm0 = COPY %15
```
where the size of `%xmm0` is 32, 64 or 128 bits, depending on context.
The function is `double`, so here 64 bits is the intended width.
Apparently, in the *.ext.uni, it was read as 128 bits.
Maybe the fix is to expand the `COPY` into some `(define) + (combine)` combo.

**Comment by matsc-at-sics-se** [comment]
Closed by commits up to commit b06903230226a10cfb31e18f953589d848f5e111.
